worlds/alttp/ItemPool.py

```diff
@@ -566,7 +571,7 @@ def place_item(loc, item):
         pool.append('Rupees (20)')
     want_progressives = world.progressive[player].want_progressives
 
-    if want_progressives(world.random):
+    if want_progressives(world.random) or progressive_gloves:
```

Comment:
If we do keep the general Progressives option, these evaluations should be swapped so that it doesn't call want_progressives unnecessarily